enigma-core/enclave/src/lib.rs

```diff
@@ -352,6 +348,7 @@ unsafe fn ecall_execute_internal(pre_execution_data: &mut Vec<Box<[u8]>>, byteco
     // Signing: S(exeCodeHash, inputsHash, delta(X-1)Hash, deltaXHash, outputHash, usedGas, optionalEthereumData, Success)
     let used_gas = result.used_gas.to_be_bytes();
     let output_hash = encrypted_output.keccak256();
+    let delta_hash = save_enc_delta(db_ptr, &exec_res.state_delta)?;
```

Comment:
I think that there is no difference between saving state and saving delta in the aspect of this PR. Therefore, I suggest (1) to split creation of delta for the signing and saving of delta; (2) implement saving of delta and state in a manner that if one fails, the other fails as well.

Reply:
I implemented this, however, there can be a scenario in which the delta is stored and the state isn't. in this situation, I can remove the delta from the DB but it seems to me like a bit of an overkill. what do you say?

Reply:
It depends on how hard the removing of delta form DB is. If the task is easy, it is better to implement it now. If it is hard of time consuming, maybe the best option is to address this in the comment and put this task in a backlog.

Reply:
this case will corrupt the DB forever, therefore, I added a todo for this, and it will be implemented once we add the option for reverting the state.